yjit/src/backend/arm64/mod.rs

```diff
@@ -1020,21 +1041,23 @@ impl Assembler
             assert!(label_idx == idx);
         }
 
-        let start_write_pos = cb.get_write_pos();
+        let start_ptr = cb.get_write_ptr();
         let gc_offsets = asm.arm64_emit(cb);
 
-        if !cb.has_dropped_bytes() {
+        if cb.has_dropped_bytes() {
+            cb.clear_labels();
```

Comment:
I think we also need to replicate these to the x86 backend

Reply:
Good catch, thanks https://github.com/ruby/ruby/pull/6460/commits/16c630a94b088202c7debb2c4afa44d1c0ad70ef